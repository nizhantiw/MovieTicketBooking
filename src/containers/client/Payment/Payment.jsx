import React, { useState } from "react";
import './Payment.scss'

const Form = () => {
  const [paymentData, setpaymentData] = useState({
    bankName: "",
    accNumber: "",
    IFSC: "",
    CVV: ""
  });

  const updatepaymentData = event =>
    setpaymentData({
      ...paymentData,
      [event.target.name]: event.target.value
    });

    const handleFormSubmit=(event)=> {
        alert("Payment Successful!!");
        event.preventDefault();
      }

  const { bankName, accNumber, IFSC, CVV } = paymentData;

  return (<div class="payment">
    <h3>Provide data to make payment </h3><br/>
    <form class="submit" onSubmit={e => handleFormSubmit(e)} id="btn">
     <div>
     <label htmlFor="InputValue">Bank Name</label><br/>
      <input id="InputValue" label="Message"
        value={bankName}
        onChange={e => updatepaymentData(e)}
        placeholder="Bank Name"
        type="text"
        name="bankName"
        required
      /></div><br/>
      <div>
      <label htmlFor="InputValue2">Account Number</label><br/>
        <input id="InputValue2"
        value={accNumber}
        onChange={e => updatepaymentData(e)}
        placeholder="Account Number"
        type="text"
        name="accNumber"
        required
      /></div><br/>
      <div>
      <label htmlFor="InputValue3">IFSC</label><br/>
        <input id="InputValue3"
        value={IFSC}
        onChange={e => updatepaymentData(e)}
        placeholder="IFSC address"
        type="IFSC"
        name="IFSC"
        required
      /></div><br/>
      <div>
      <label htmlFor="InputValue4">CVV</label><br/>
        <input id="InputValue4"
        value={CVV}
        onChange={e => updatepaymentData(e)}
        placeholder="CVV"
        type="CVV"
        name="CVV"
        required
      /></div><br/>

<button id="submit" type="submit" value="Submit" >Submit</button>
    </form>
    </div>
  );
};

export default Form;
