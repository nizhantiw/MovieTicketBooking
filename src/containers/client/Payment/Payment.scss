.body{
    background-color: blue;
}

.payment{
    background-color: lightcyan;
    height: 100vh;
}