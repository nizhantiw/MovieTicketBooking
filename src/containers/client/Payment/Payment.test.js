import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount  } from 'enzyme';
import renderer from 'react-test-renderer'
import Payment from './Payment'
import { render, fireEvent } from "@testing-library/react";



configure({adapter: new Adapter()});

it("renders correctly", () => {
  const wrapper = shallow(
    <Payment />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders snapshot', () => {
  const tree = renderer.create(<Payment/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render button', () => {
  const wrapper = shallow(<Payment />)
  const buttonElement  = wrapper.find('#submit');
  expect(buttonElement).toHaveLength(1);
  expect(buttonElement.text()).toEqual('Submit');
})


it("show the sended message", () => {
  const onSubmit = jest.fn();
  const { getByLabelText, getByText } = render(<Payment />);
  const inputValue = "abc";
  const inputValue2 = "abc";
  const inputValue3 = "abc";
  const inputValue4 = "abc";

  fireEvent.change(getByLabelText("Bank Name"), { target: { value: inputValue } });
  fireEvent.change(getByLabelText("Account Number"), { target: { value: inputValue2 } });
  fireEvent.change(getByLabelText("IFSC"), { target: { value: inputValue3 } });
  fireEvent.change(getByLabelText("CVV"), { target: { value: inputValue4 } });
  fireEvent.click(getByText(/Submit/i));


  //expect(onSubmit).toBeCalled();
  expect(onSubmit).toHaveBeenCalledTimes(0);
});