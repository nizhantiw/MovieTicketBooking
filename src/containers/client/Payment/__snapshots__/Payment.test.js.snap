// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders snapshot 1`] = `
<div
  class="payment"
>
  <h3>
    Provide data to make payment 
  </h3>
  <br />
  <form
    class="submit"
    id="btn"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="InputValue"
      >
        Bank Name
      </label>
      <input
        id="InputValue"
        label="Message"
        name="bankName"
        onChange={[Function]}
        placeholder="Bank Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="InputValue2"
      >
        Account Number
      </label>
      <input
        id="InputValue2"
        name="accNumber"
        onChange={[Function]}
        placeholder="Account Number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="InputValue3"
      >
        IFSC
      </label>
      <input
        id="InputValue3"
        name="IFSC"
        onChange={[Function]}
        placeholder="IFSC address"
        required={true}
        type="IFSC"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="InputValue4"
      >
        CVV
      </label>
      <input
        id="InputValue4"
        name="CVV"
        onChange={[Function]}
        placeholder="CVV"
        required={true}
        type="CVV"
        value=""
      />
    </div>
    <br />
    <button
      id="submit"
      type="submit"
      value="Submit"
    >
      Submit
    </button>
  </form>
</div>
`;
